### PR TITLE
Cleanup of "when" statements no longer supporting old-style implicit captures

### PR DESCRIPTION
**Pull request by vpax**
Now that https://github.com/zeek/zeek/pull/2881 has been merged, I've gone ahead and trimmed a bunch of code no longer needed due to the removal of old-style implicit captures for `when` statements.  Along the way I transformed the `function_ingredients` struct to a class (`FunctionIngredients`) because it had grown pretty heavy-weight for a struct.  I also fixed up printing of new-style `when` statements to use their original form rather than the lambda-transformed internals, streamlined the `Trigger` class and the `Frame` cloning it uses, and removed the nascent `when` support in ZAM (which never actually got built out).

### CODE REVIEW

File: src/Frame.cc [hunk 1]
```diff
@@ -120,6 +120,17 @@ Frame* Frame::Clone() const
 	return other;
 	}
 
+Frame* Frame::LightClone() const
+	{
+	Frame* other = new Frame(0, function, func_args);
```
[the remark this answers is not in the document]
Comment (awelzel) [reply src/Frame.cc:125]: Hmm, maybe `TriggerClone()` or `CloneForTrigger()` given it's used only there?

**Review by awelzel — APPROVED** [review]
Hadn't realized there's so much more to ditch.

Just a few nits. I'm happy to move this in after one more round.

File: src/Frame.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (vpax) [reply src/Frame.cc:125]: I like `CloneForTrigger()`.  Feel free to go ahead and make that change; or I'm happy to do so if you prefer.

File: src/Stmt.cc [hunk 2]
```diff
@@ -1996,38 +1974,30 @@ void WhenInfo::Build(StmtPtr ws)
 
 	auto shebang = make_intrusive<StmtList>(do_test, do_bodies, dummy_return);
 
-	auto ingredients = std::make_unique<function_ingredients>(current_scope(), shebang,
-	                                                          current_module);
-	auto outer_ids = gather_outer_ids(pop_scope(), ingredients->body);
+	auto ingredients = std::make_unique<FunctionIngredients>(current_scope(), shebang,
```
[the remark this answers is not in the document]
Comment (vpax) [reply src/Stmt.cc:1977]: Ah - you mean that it got folded into a different commit, right?  Yeah, that was me slightly cutting corners when bundling up everything after I got it working.

File: src/Frame.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (awelzel) [reply src/Frame.cc:125]: I'll pick it up tomorrow.

**Pull request merged**